sql-plugin/src/main/scala/com/nvidia/spark/rapids/Plugin.scala

```diff
@@ -305,11 +309,14 @@ object ExecutionPlanCaptureCallback {
   }
 
   private def didFallBack(plan: SparkPlan, fallbackCpuClass: String): Boolean = {
-    if (!plan.isInstanceOf[GpuExec] &&
-      getBaseNameFromClass(plan.getClass.getName) == fallbackCpuClass) {
-      true
-    } else {
-      plan.expressions.exists(didFallBack(_, fallbackCpuClass))
+    plan match {
+      case p: AdaptiveSparkPlanExec => didFallBack(p.executedPlan, fallbackCpuClass)
+      case _ => if (!plan.isInstanceOf[GpuExec] &&
```

Comment:
The match should be leveraged to do the instance checking here, e.g.:
```scala
plan match {
  case p: AdaptiveSparkPlanExec => ...
  case p: GpuExec => getBaseNameFromClass(p.getClass.getName) == fallbackCpuClass
  case p => p.expressions.exists(didFallBack(p, fallbackCpuClass)
}
```

Reply:
Fixed.